jsonrpc2/http.go

```diff
@@ -131,8 +131,11 @@ func (conn *httpClientConn) Write(buf []byte) (int, error) {
 			var resp *http.Response
 			resp, err = conn.doer.Do(req)
 			const maxBodySlurpSize = 32 * 1024
+
+			mediaType, _, _ := mime.ParseMediaType(resp.Header.Get("Content-Type"))
```

Comment:
At this point (before `if err != nil`) `resp` can be `nil`. Please fix this.

And add test for this case, please.